x/bridge/keeper/assets_locked_test.go

```diff
@@ -205,6 +205,102 @@ func TestAcceptAssetsLocked(t *testing.T) {
 			},
 			errContains: "failed to send coins",
 		},
+		{
+			name: "BTC mint to blocked address as recipient",
+			bankKeeperFn: func(ctx sdk.Context) *mockBankKeeper {
+				bankKeeper := newMockBankKeeper()
+
+				bankKeeper.On(
+					"MintCoins",
+					ctx,
+					types.ModuleName,
+					sdk.NewCoins(
+						sdk.NewCoin(evmtypes.DefaultEVMDenom, math.NewInt(1)),
+					),
+				).Return(nil)
+
+				bankKeeper.On(
+					"SendCoinsFromModuleToAccount",
+					ctx,
+					types.ModuleName,
+					mock.Anything,
+					mock.Anything,
+				).Return(nil)
+
+				return bankKeeper
+			},
+			evmKeeperFn: func(_ sdk.Context) *mockEvmKeeper {
+				return newMockEvmKeeper()
+			},
+			events: types.AssetsLockedEvents{
+				// Sequence tip is 10, so the expected start is 11.
+				mockEvent(11, testBlockedAddress, 1, testSourceBTCToken),
+			},
+			errContains: "", // This case shouldn't lead to an error.
+			postCheckFn: func(ctx sdk.Context, k Keeper) {
+				// The bank state change should not have been executed.
+				k.bankKeeper.(*mockBankKeeper).AssertNotCalled(
+					t,
+					"MintCoins",
+					mock.Anything,
+					mock.Anything,
+				)
+
+				// The bank state change should not have been executed.
+				k.bankKeeper.(*mockBankKeeper).AssertNotCalled(
+					t,
+					"SendCoinsFromModuleToAccount",
+					mock.Anything,
+					mock.Anything,
+				)
+
+				// The sequence tip should have been updated as the event
+				// was skipped.
+				require.EqualValues(
+					t,
+					math.NewInt(11),
+					k.GetAssetsLockedSequenceTip(ctx),
+				)
+			},
+		},
+		{
+			name:         "ERC20 using blocked address as recipient",
+			bankKeeperFn: func(_ sdk.Context) *mockBankKeeper { return newMockBankKeeper() },
+			evmKeeperFn: func(ctx sdk.Context) *mockEvmKeeper {
+				evmKeeper := newMockEvmKeeper()
+
+				evmKeeper.On(
+					"ExecuteContractCall",
+					ctx,
+					mock.Anything,
+				).Return(&evmtypes.MsgEthereumTxResponse{}, nil)
+
+				return evmKeeper
+			},
+			events: types.AssetsLockedEvents{
+				// Sequence tip is 10, so the expected start is 11.
+				// Use an arbitrary token whose mapping is not set.
+				mockEvent(11, testBlockedAddress, 1, "0x96c1493e5a9efa5C7572D98b5e9544B17bD9AE72"),
```

Comment:
If we use a not mapped token, we don't really know why the bridge rejected this event and we don't test the intended behavior precisely. We need to ensure blocked address is the only cause of rejection here.

Reply:
addressed in: 34c05ba4c2ca492883883a4a4489e646382cdea4